File: ch07/CSharpServerControl1/Listing07-31.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CSharpServerControl1
{
    [DefaultProperty("Text")]
    [ToolboxData("<{0}:Listing0731 runat=server></{0}:Listing0731>")]
    public class Listing0731 : CompositeControl
    {
        protected TextBox textbox = new TextBox();

        protected override void CreateChildControls()
        {
            this.Controls.Add(textbox);
        }
    }
}

File: ch07/VbServerControl1/Listing07-31.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Text
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls

<DefaultProperty("Text"),
 ToolboxData("<{0}:Listing0731 runat=server></{0}:Listing0731>")>
Public Class Listing0731
    Inherits System.Web.UI.WebControls.CompositeControl

    Protected textbox As TextBox = New TextBox()

    Protected Overrides Sub CreateChildControls()
        Me.Controls.Add(textbox)
    End Sub
End Class